### PR TITLE
add javascript "Loops" section

### PR DESCRIPTION
**Pull request by dotcode**

### CODE REVIEW

File: languages/javascript.md [hunk 1]
```diff
@@ -432,6 +433,36 @@ else {
...
+#### Loops
+
+In `for` loops, there must be space between the semi-colons.
+
+`for..in` loops must include a check with `hasOwnProperty`.
+
+We do this:
+
+```js
+for (var i; i < 10; i += 1) {
```
Comment (danyalaytekin) [inline languages/javascript.md]: `var i = 0`

File: languages/javascript.md [hunk 2]
```diff
@@ -432,6 +433,36 @@ else {
...
+    if (obj.hasOwnProperty(prop)) {
+        // ...
+    }
+}
+```
+
+We _don't_ do this:
+
+```js
+for (var i;i < 10;i += 1) // ...
```
Comment (danyalaytekin) [inline languages/javascript.md]: As 445

File: languages/javascript.md (hunk 1 above)
Comment (hollsk) [inline languages/javascript.md]: "there must be a space _after_ the semi-colons", surely, since the marks themselves are not sequential

**Comment by hollsk** [comment]
👍

**Comment by danyalaytekin** [comment]
:+1:

**Pull request merged**